### PR TITLE
First, but working, version

### PR DESCRIPTION
**Pull request by stefankleff**
-working with phpredis / adapter for predis missing
-working with zend redis resource manager / standalone factories missing
-No tests yet
-Coding style not checked yet
-Some method signatures missing
-recover() not tested

### CODE REVIEW

**Comment by juriansluiman** [comment]
:+1: :kissing_closed_eyes:

Will have a look this weekend, but doing this work is already great! Is it OK to comment on the PR and refine where needed, or do you prefer a faster merge and additional PRs to refine? (I'm for the first option, btw)

File: Module.php [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    Feature\AutoloaderProviderInterface,
+    Feature\ConfigProviderInterface,
+    Feature\ConsoleBannerProviderInterface,
+    Feature\ConsoleUsageProviderInterface,
+    Feature\DependencyIndicatorInterface
+{
+    /**
+     * {@inheritDoc}
+     */
+    public function getAutoloaderConfig()
```
Comment (bakura10) [inline Module.php:22]: Do we still need this (considering removing it for all other SlmQueue\* also)

File: config/module.config.php [hunk 2]
```diff
@@ -0,0 +1,66 @@
+<?php
+
+return array(
+    'service_manager' => array(
+        'factories' => array(
+            'SlmQueueRedis\Worker\RedisWorker' => 'SlmQueue\Factory\WorkerFactory',
```
Comment (bakura10) [inline config/module.config.php:6]: Alignment

File: src/SlmQueueRedis/Adapter/AbstractRedisAdapter.php [hunk 3]
```diff
@@ -0,0 +1,45 @@
+<?php
+namespace SlmQueueRedis\Adapter;
+
+abstract class AbstractRedisAdapter implements AdapterInterface {
```
Comment (bakura10) [inline src/SlmQueueRedis/Adapter/AbstractRedisAdapter.php:4]: Tons of CS fixes here. I won't do it right now though :p.

File: src/SlmQueueRedis/Adapter/PhpRedisAdapter.php [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+    /**
+     * @param string $queue
+     * @param int $leaseTime
+     * @param int $timeout
+     * @return null|array
+     */
+    public function pop($queue, $leaseTime = 3600, $timeout = RedisQueue::BLOCKING_DISABLED)
+    {
+        // get next id from pending list and move id to working list
+        if($timeout == RedisQueue::BLOCKING_DISABLED) {
```
Comment (bakura10) [inline src/SlmQueueRedis/Adapter/PhpRedisAdapter.php]: Should be === ?

File: src/SlmQueueRedis/Adapter/PredisAdapter.php [hunk 5]
```diff
@@ -0,0 +1,11 @@
+<?php
+
+namespace SlmQueueRedis\Adapter;
+
+/**
+ * RedisQueue
+ */
+class PrredisAdapter implements AdapterInterface
```
Comment (bakura10) [inline src/SlmQueueRedis/Adapter/PredisAdapter.php]: Typo, should be Predis

**Comment by juriansluiman** [comment]
The reason I started with a `ClientInterface` instead of an `AdapterInterface`, is you interface the Redis methods, and not the queue methods. For the queue, it will simply call the required Redis commands, undifferent for the client underneath. Now, you have 50% of the queue code (serialize job and so on) in the queue and 50% of the code (the pop, push etc methods) also in the adapter.

This way, you tie the SlmQueueRedis implementation to every adapter, while the actual implementation "should" (I think) reside only in the queue. Thoughts?

**Comment by stefankleff** [comment]
Thats what I started with. But during the implementation I found no good way to abstract transactions: https://github.com/nrk/predis#transactions vs https://github.com/phpredis/phpredis#multi-exec-discard 
So I came up with abstraction on transaction level instead of command level. Sadly this results in almost the same methods as in the queue. I was close to remove the adapters and write two queue classes instead.

**Pull request closed without merging**